src/app/static/src/app/components/plots/bubble/BubblePlot.vue

```diff
@@ -140,20 +195,60 @@
                 return Math.max(...levelNums);
             },
             currentFeatures() {
-                return this.featuresByLevel[this.detail]
+                return this.featuresByLevel[this.selections.detail]
             },
             indicatorNameLookup() {
                 return toIndicatorNameLookup(this.indicators)
-            }
+            },
+            areaFilter() {
+                return this.filters.find((f:Filter) => f.id == this.areaFilterId)!!;
+            },
+            nonAreaFilters() {
+                return this.filters.filter((f: Filter) => f.id != this.areaFilterId);
+            },
+            areaFilterOptions() {
+                return this.countryFilterOption ? (this.countryFilterOption as any).children : [];
+            },
+            selectedAreaFilterOptions() {
+                const selectedOptions = this.selections.selectedFilterOptions[this.areaFilterId];
+                if (selectedOptions && selectedOptions.length > 0) {
+                    return selectedOptions
+                }
+                return this.countryFilterOption ? [this.countryFilterOption] : [];
+            },
+            flattenedAreas() {
+                return flattenOptions(this.areaFilter.options);
+            },
+            allSelectedAreaIds() {
+                //get intersection of selected areas and current Features(ie current level)
```

Comment:
I don't think it's neccessary to take this intersection since only current features are rendered on the map - you can just directly pass the flattened selected area ids into the `getFeatureIndicators` function.

Reply:
Yeah, that's right - I think I put that in there to keep the featureIndicators dictionary small. I guess it's a trade off between handling more data than you need and recalculating it more often. I don't think I saw a performance hit either way really - for Malawi at least... It would keep the bubblePlot code a little simpler to do away with allSelectedAreaIds so that's reason enough!

Reply:
I wondered if it was a deliberate choice, but yeah I think if we can simplify this a little bit that's preferable!